R/commonTTest.R

```diff
@@ -419,9 +419,19 @@ summarySEwithin <- function(data=NULL, measurevar, betweenvars=NULL, withinvars=
   ndatac <- summarySE(ndata, measurevar_n, groupvars=c(betweenvars, withinvars), na.rm=na.rm, conf.interval=conf.interval, .drop=.drop, errorBarType=errorBarType,
                       usePooledSE=usePooledSE, dependentName = dependentName, subjectName = subjectName)
 
+  # get distinct observations in a way that works for both character and factor/ ordered data
+  .nDistinctObservations <- function(x) {
+    if (is.character(x))
+      return(dplyr::n_distinct(x))
```

Comment:
Can't we just do `length(unique(x))` here? Otherwise this would necessitate adding `dplyr` to the `DESCRIPTION` (and we also already have `plyr` there, which doesn't play nicely with `dplyr`).

Reply:
I thought I checked that `dplyr` was already a dependency before using it, but apparently not!